test/common/testvm.py

```diff
@@ -847,57 +858,182 @@ def virEventLoopNativeRun():
+        self.last = 0
+
+    def _lock(self, start, step=1, force=False):
+        resources = os.path.join(tempfile.gettempdir(), ".cockpit-test-resources")
+        if not os.path.exists(resources):
+            os.mkdir(resources, 0755)
+        for port in range(start, start + (100 * step), step):
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            lockpath = os.path.join(resources, "network-{0}".format(port))
+            try:
+                lockf = os.open(lockpath, os.O_WRONLY | os.O_CREAT)
+                fcntl.flock(lockf, fcntl.LOCK_NB | fcntl.LOCK_EX)
+                sock.bind(("127.0.0.1", port))
+                self.locked.append(lockf)
+            except IOError:
+                if force:
+                    return port
+                os.close(lockf)
+                continue
+            else:
+                return port
+            finally:
+                sock.close()
+        raise Failure("Couldn't find unique network port number")
+
+    # Create resources for an interface, returns address and XML
+    def interface(self, number=None):
+        if number is None:
+            number = self.last + 1
+        if number > self.last:
+            self.last = number
+        mac = self._lock(10000 + self.offset + number) - (10000 + self.offset)
+        result = {
+            "number": self.offset + number,
+            "mac": 'ee:00:00:{:02x}:{:02x}:{:02x}'.format(mac & 0xff0000 >> 24, mac & 0xff00 >> 16, mac & 0xff),
+            "name": "m{0}.cockpit.lan".format(mac)
+        }
+        return result
+
+    # Create resources for a host, returns address and XML
+    def host(self, number=None, restrict=False):
+        result = self.interface(number)
+        result["mcast"] = self.network
+        result["restrict"] = restrict and "on" or "off"
+        result["control"] = "127.0.0.2:{0}".format(self._lock(2200 + result["number"]))
+        result["browser"] = "127.0.0.2:{0}".format(self._lock(9090 + result["number"]))
+        if self.bridge:
+            result["bridge"] = self.bridge
+            result["networks"] = TEST_BRIDGE_XML.format(**result)
+        else:
+            result["bridge"] = ""
+            result["networks"] = TEST_MCAST_XML.format(**result)
+        result["qemu"] = TEST_REDIR_XML.format(**result)
+        return result
+
+    def kill(self):
+        locked = self.locked
+        self.locked = [ ]
+        for x in locked:
+            os.close(x)
+
 class VirtMachine(Machine):
+    network = None
     memory_mb = None
     cpus = None
 
-    def __init__(self, **args):
+    def __init__(self, networking=None, maintain=False, memory_mb=None, cpus=None, **args):
+        self.maintain = maintain
+
+        # Currently all images are x86_64. When that changes we will have
+        # an override file for those images that are not
+        self.arch = "x86_64"
```

Comment:
This is misleading -- we do have an "i386" image now. It can of course run on a 64 bit (virtual) machine. Maybe for now change the comment to "all images run on x86_64"?